test/gtest/uct/ib/test_ud.cc

```diff
@@ -570,20 +568,27 @@ UCS_TEST_SKIP_COND_P(test_ud, ca_md,
 
     ep(m_e1)->tx.tx_hook = count_tx;
     do {
+        uct_ud_enter(iface(m_e1));
```

Comment:
why need to protect it?

Reply:
in case ud async progress does re transmissions and changes tx_count and/or cwnd

Reply:
afaiu, it should not affect test flow anyway
not that improtant comment though

Reply:
in theory, the async thread can do re-transmissions and reduce window size at the same time the main thread is reading tx_counf and cwnd,  on CPUs with weak mem ordering the main thread could see the "new" value of tx_count, but prev value of cwnd, so will fail the test because the window will shrink but nothing new will be sent